BSPConvert.Lib/Source/EntityConverter.cs

```diff
@@ -597,23 +612,192 @@ private void ConvertTriggerTargetsRecursive(Entity trigger, Entity entity, float
+
+			if (!float.TryParse(targetScore["count"], out var count))
+				count = 1;
+
+			ModifyMathCounter(entity, output, "Add", $"{count}", delay);
+		}
+
+		private Entity CreateLogicCase()
+		{
+			var logicCase = new Entity();
+			logicCase.ClassName = "logic_case";
+			logicCase.Name = "logiccase";
+
+			for (var i = 1; i <= 16; i++) // Logic_case supports 16 different outputs
+			{
+				var caseNum = $"case{i:D2}";
+				logicCase[$"{caseNum}"] = $"{i}";
 			}
+
+			var connection = new Entity.EntityConnection()
+			{
+				name = "OnUsed",
+				target = "*relay*", // Disable all logic_relays
+				action = "Disable",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			logicCase.connections.Add(connection);
+
+			sourceEntities.Add(logicCase);
+
+			return logicCase;
+		}
+
+		private void CreateMathCounter()
+		{
+			var counter = new Entity();
+			counter.ClassName = "math_counter";
+			counter.Name = "mathcounter";
+			counter["startvalue"] = "0";
+			counter["min"] = "0";
+			counter["max"] = "16";
+
+			var connection = new Entity.EntityConnection()
+			{
+				name = "OutValue",
+				target = "logiccase",
+				action = "InValue",
+				param = null,
+				delay = 0,
+				fireOnce = -1
+			};
+			counter.connections.Add(connection);
+
+			sourceEntities.Add(counter);
+		}
+
+		private void ConvertFragsFilter(Entity entity, Entity targetFragsFilter, string output, float delay)
+		{
+			if (!int.TryParse(targetFragsFilter["frags"], out var frags))
+				frags = 1; // Default number of frags is 1 if no value is specified
+
+			targetFragsFilter.ClassName = "logic_relay";
+			targetFragsFilter["startdisabled"] = (frags > 0) ? "1" : "0"; // Players start with 0 frags, disable entities that require > 0
+
+			targetFragsFilter.Name += $"relay{frags:D2}"; // Name needs a unique relay number for the logic_case to target
+
+			var connection = new Entity.EntityConnection()
+			{
+				name = output,
+				target = targetFragsFilter.Name,
+				action = "Trigger",
+				param = null,
+				delay = delay,
+				fireOnce = -1
+			};
+			entity.connections.Add(connection);
+
+			var match = false;
+			var spawnflags = (TargetFragsFilterFlags)targetFragsFilter.Spawnflags;
+
+			if (spawnflags.HasFlag(TargetFragsFilterFlags.Reset)) // Reset frags to 0
+				ModifyMathCounter(targetFragsFilter, "OnTrigger", "SetValue", "0", delay);
+			else if (spawnflags.HasFlag(TargetFragsFilterFlags.Remover)) // Remove frags when used
+				ModifyMathCounter(targetFragsFilter, "OnTrigger", "Subtract", $"{frags}", delay);
+
+			if (spawnflags.HasFlag(TargetFragsFilterFlags.Match))
+				match = true;
+
+			AddLogicCaseOutput(targetFragsFilter.Name, frags, match);
+
+			ConvertTriggerTargetsRecursive(targetFragsFilter, targetFragsFilter, 0, new HashSet<Entity>());
+		}
+
+		private void AddLogicCaseOutput(string targetName, int frags, bool match)
+		{
+			var logicCase = sourceEntities.FirstOrDefault(x => x.ClassName == "logic_case") ?? CreateLogicCase();
```

Comment:
I think `sourceEntities.Find` is slightly more performant and does the same thing.